### PR TITLE
Issue 220.dockter.1

### PR DESCRIPTION
**Pull request by docktermj**
# Pull request questions

## Which issue does this address

Issue number: #220

### CODE REVIEW

File: stream-loader.py [hunk 1]
```diff
@@ -1371,87 +1391,160 @@ def update_active_g2_configuration(self):
...
         except Exception as err:
-            if self.is_g2_default_configuration_changed():
-                self.update_active_g2_configuration()
-                try:
-                    self.g2_engine.addRecord(data_source, record_id, jsonline)
-                except Exception as err:
-                    raise err
-            else:
-                raise err
+            raise err
```
Comment (jeffadair) [inline stream-loader.py:1408]: Can we get rid of this try/except block if we are just reraising the exception and nothing else? (This also applies to the other process_* functions.)

File: stream-loader.py [hunk 2]
```diff
@@ -1371,87 +1391,160 @@ def update_active_g2_configuration(self):
...
+
+            # Put "info" on info queue.
+
+            if filtered_response_json:
+                self.add_to_info_queue(filtered_response_json)
+                logging.debug(message_debug(904, threading.current_thread().name, filtered_response_json))
+
+        logging.debug(message_debug(951, sys._getframe().f_code.co_name))
+
+    def send_jsonline_to_g2_engine(self, jsonline, senzing_stream_loader_value_default={"action": 'addRecord'}):
```
Comment (jeffadair) [inline stream-loader.py:1547]: Clever! I don't think I would have thought of using an actual dictionary as the default for senzing_stream_loader_value_default. But it makes the code a bit more clean.

File: stream-loader.py [hunk 3]
```diff
@@ -389,6 +389,11 @@
         "env": "SENZING_SQS_WAIT_TIME_SECONDS",
         "cli": "sqs-wait-time-seconds"
     },
+    "stream_loader_directive": {
+        "default": "senzingStreamLoader",
+        "env": "SENZING_STREAM_LOADER_DIRECTIVE",
```
[the remark this answers is not in the document]
Comment (docktermj) [reply stream-loader.py:394]: I like your observation.   I keep bouncing between `SENZING_STREAM_LOADER_DIRECTIVE_KEY`  as in key/value   or  `SENZING_STREAM_LOADER_DIRECTIVE_NAME` as in the  "property", "property name", "property value" language used in Google's [JSON Style Guide](https://google.github.io/styleguide/jsoncstyleguide.xml)   Any thoughts?

File: README.md [hunk 4]
```diff
@@ -412,6 +413,37 @@ Unset `*_PARAMETER` environment variables have no effect on the
...
+    {
+        "action": "<action-identifier>"
+    }
+    ```
+
+1. The supported "action-identifiers" are:
+    1. [addRecord](https://docs.senzing.com/python/senzing-G2Engine-reference.html#addRecord)
+    1. [addRecordWithInfo](https://docs.senzing.com/python/senzing-G2Engine-reference.html#addRecordWithInfo)
+    1. [reevaluateRecord](https://docs.senzing.com/python/senzing-G2Engine-reference.html#reevaluateRecord)
+    1. [reevaluateRecordWithInfo](https://docs.senzing.com/python/senzing-G2Engine-reference.html#reevaluateRecordWithInfo)
```
Comment (roncewind) [inline README.md:434]: did we decide not to support re-evaluate?

File: stream-loader.py [hunk 5]
```diff
@@ -1371,87 +1391,139 @@ def update_active_g2_configuration(self):
...
+
+            # Put "info" on info queue.
+
+            if filtered_response_json:
+                self.add_to_info_queue(filtered_response_json)
+                logging.debug(message_debug(904, threading.current_thread().name, filtered_response_json))
+
+        logging.debug(message_debug(951, sys._getframe().f_code.co_name))
+
+    def process_reevaluateRecord(self, message_metadata, message_dict):
```
Comment (roncewind) [inline stream-loader.py:1482]: re-evaluate?  are we supporting this?

File: README.md (hunk 4 above)
Comment (docktermj) [reply README.md:434]: There seems to be 2 types of "reevaluate"   One is reevaluateRecord, the other is reevaluateEntity.   So the discussion was on not implementing reevaluatingEntity.

File: stream-loader.py (hunk 5 above)
Comment (docktermj) [reply stream-loader.py:1482]: reevaluateRecords is the one (of the two) that we are supporting.

**Review by roncewind — APPROVED** [review]

**Pull request merged**